### PR TITLE
Bug changes for today's testing

### PR DESCRIPTION
**Pull request by alicetouhou**
every one found so far should be fixed
although i am unsure about making a few cards not run onApply hooks when moving, may need to ask sense what they think first. i think i can use a different solution for biffa if this doesn't work.

### CODE REVIEW

File: common/cards/single-use/lantern.ts [hunk 1]
```diff
@@ -40,9 +41,17 @@ class LanternSingleUseCard extends SingleUseCard {
...
-					player.pile.filter((c) => !modalResult.cards.includes(c))
-					modalResult.cards.reverse().map((c: CardT) => player.hand.push(c))
+					const cards: Array<CardT> = modalResult.cards
+
+					player.pile = player.pile.filter((c) => {
+						if (cards.some((d) => c.cardInstance === d.cardInstance)) return false
+						return true
+					})
+
+					cards.map((c) => player.hand.push(c))
```
Comment (Tyrannicodin) [inline common/cards/single-use/lantern.ts]: This should be forEach
```suggestion
					cards.forEach((c) => player.hand.push(c))
```

File: common/cards/single-use/lantern.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import {CardPosModel} from '../../models/card-pos-model'
 import {GameModel} from '../../models/game-model'
 import {CardT} from '../../types/game-state'
+import { drawCards } from '../../utils/movement'
```
Comment (Tyrannicodin) [inline common/cards/single-use/lantern.ts:4]: Formatting
```suggestion
import {drawCards} from '../../utils/movement'
```

File: common/cards/single-use/glowstone.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import {CardPosModel} from '../../models/card-pos-model'
 import {GameModel} from '../../models/game-model'
 import { CardT } from '../../types/game-state'
+import { drawCards } from '../../utils/movement'
```
Comment (Tyrannicodin) [inline common/cards/single-use/glowstone.ts]: Formatting
```suggestion
import {drawCards} from '../../utils/movement'
```

**Comment by alicetouhou** [comment]
i ended up implementing the different (better) solution

File: common/cards/single-use/glowstone.ts [hunk 4]
```diff
@@ -37,13 +38,18 @@ class GlowstoneSingleUseCard extends SingleUseCard {
...
+						if (cards.some((d) => c.cardInstance === d.cardInstance)) return false
+						return true
+					})
 
-					const selectedCards = opponentPlayer.pile.slice(0,3).filter((c) => c !== cards[0])
-					opponentPlayer.pile.filter((c) => !selectedCards.includes(c))
-					selectedCards.map((c) => opponentPlayer.pile.unshift(c))
-					opponentPlayer.hand.push(cards[0])
+					opponentPlayer.pile = opponentPlayer.pile.slice(3)
+					bottomCards.map((c) => opponentPlayer.pile.push(c))
```
Comment (Tyrannicodin) [inline common/cards/single-use/glowstone.ts]: This map and the one below should be forEach

File: common/ailments/museum-collection.ts [hunk 5]
```diff
@@ -18,15 +18,24 @@ class MuseumCollectionAilment extends Ailment{
...
 	override onApply(game: GameModel, ailmentInfo: AilmentT, pos: CardPosModel) {
 		game.state.ailments.push(ailmentInfo)
+		const oldHandSize = this.getInstanceKey(ailmentInfo.ailmentInstance)
 		const {player} = pos
 
-		player.hooks.onAttach.add(ailmentInfo.ailmentInstance, () => {
+		player.custom[oldHandSize] = player.hand.length
+
+		player.hooks.onAttach.add(ailmentInfo.ailmentInstance, (instance) => {
+			console.log("Old: " + player.custom[oldHandSize] + " | New: " + player.hand.length)
```
Comment (Tyrannicodin) [inline common/ailments/museum-collection.ts]: This shouldn't be here

File: common/cards/single-use/brush.ts [hunk 6]
```diff
@@ -40,11 +40,17 @@ class BrushSingleUseCard extends SingleUseCard {
...
 
 					const cards: Array<CardT> = modalResult.cards
+					const bottomCards: Array<CardT> = player.pile.slice(0, 3).filter((c) => {
+						if (cards.some((d) => c.cardInstance === d.cardInstance)) return false
+						return true
+					})
 
-					player.pile.filter((c) => !cards.includes(c))
+					player.pile = player.pile.slice(3)
 					cards.reverse().map((c) => player.pile.unshift(c))
```
Comment (Tyrannicodin) [inline common/cards/single-use/brush.ts]: forEach here

**Review by Tyrannicodin — CHANGES_REQUESTED** [review]
A log + some formatting and maps that should be forEach

**Pull request merged**